qadence/extensions.py

```diff
@@ -2,41 +2,72 @@
 
 import importlib
 from string import Template
+from typing import TypeVar
 
-from qadence.backend import Backend
+from qadence.backend import Backend, BackendConfiguration
 from qadence.blocks.abstract import TAbstractBlock
+from qadence.engines.differentiable_backend import DifferentiableBackend
 from qadence.logger import get_logger
 from qadence.types import BackendName, DiffMode, Engine
 
 backends_namespace = Template("qadence.backends.$name")
+BackendClsType = TypeVar("BackendClsType", bound=Backend)
+EngineClsType = TypeVar("EngineClsType", bound=DifferentiableBackend)
 
 logger = get_logger(__name__)
 
 
-def _available_engines() -> dict:
+def import_config(backend_name: str | BackendName) -> BackendConfiguration:
+    module_path = f"qadence.backends.{backend_name}.config"
```

Comment:
We should probably think of a better way to handle module paths than to hardcode them. Maybe we can use templated strings here ?